lib/bugsnag/middleware/delayed_job.rb

```diff
@@ -0,0 +1,71 @@
+module Bugsnag::Middleware
+  ##
+  # Attaches delayed_job information to an error report
+  class DelayedJob
+    def initialize(bugsnag)
+      @bugsnag = bugsnag
+    end
+
+    def call(report)
+      job = report.request_data[:delayed_job]
+      if job
+        job_data = {
+          :class => job.class.name,
+          :id => job.id
+        }
+        job_data[:priority] = job.priority if job.respond_to?(:priority)
+        job_data[:run_at] = job.run_at if job.respond_to?(:run_at)
+        job_data[:locked_at] = job.locked_at if job.respond_to?(:locked_at)
+        job_data[:locked_by] = job.locked_by if job.respond_to?(:locked_by)
+        job_data[:created_at] = job.created_at if job.respond_to?(:created_at)
+        job_data[:queue] = job.queue if job.respond_to?(:queue)
+
+        if job.respond_to?('payload_object')
+          job_data[:active_job] = job.payload_object.job_data if job.payload_object.respond_to?('job_data')
+          job_data[:payload] = construct_job_payload(job.payload_object)
+        end
+
+        if job.respond_to?(:attempts)
+          max_attempts = (job.respond_to?(:max_attempts) && job.max_attempts) || Delayed::Worker.max_attempts
+          job_data[:attempts] = "#{job.attempts + 1} / #{max_attempts}"
+          # +1 as "attempts" is zero-based and does not include the current failed attempt
```

Comment:
Would it make sense to have the DelayedJob integration/plugin register an after callback on the `:error` event instead? It seems more semantically appropriate, and I think there wouldn't be a need to add one to `job.attempts`.

Reply:
From doing some investigation the method we're currently using seems to be the most common method of adding error handling to all jobs within an application without disrupting error handling on specific jobs or workers.  However if there is an alternative, it would be good to review it while these changes are being made.